### PR TITLE
Snowflake SSO with G-suite

### PR DESCRIPTION
**Pull request by zmaleki**

### CODE REVIEW

File: _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+image: img/blog/snowflake-g-suite/heading.jpg
+---
+
+<center><img src="/img/blog/snowflake-g-suite/heading.jpg" /></center><br/>
+
+Recently I was trying to connect snowflake to SSO with G-suite and I went through a world of pain to get it working. Documentations were out of date and did not help with 403 errors that I was continuously receiving. Therefore, I decided to write up this blogpost, hoping it would help someone in future. <br/>
+
+# G-Suite Setup<br/>
+
+To get started you first need to create a new SAML app in G-suit. You may require G-Suite admin access to perform this action:<br/>
```
Comment (paulmech) [inline _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md:16]: `G-Suit` => `G-Suite`

File: _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+title: Snowflake SSO setup with G-Suite
+date: 2021-07-5
+tags: [snowflake saml sso gsuite google authentication]
+author: Zainab Maleki
+image: img/blog/snowflake-g-suite/heading.jpg
+---
+
+<center><img src="/img/blog/snowflake-g-suite/heading.jpg" /></center><br/>
+
+Recently I was trying to connect snowflake to SSO with G-suite and I went through a world of pain to get it working. Documentations were out of date and did not help with 403 errors that I was continuously receiving. Therefore, I decided to write up this blogpost, hoping it would help someone in future. <br/>
```
Comment (Hamo69) [inline _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md:12]: > snowflake
to Snowflake

Comment (Hamo69) [inline _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md:12]: > Documentations were out of date 
to The documentation was out of date

File: _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+```sql
+select t.value:type::varchar as type,
+       t.value:host::varchar as host,
+       t.value:port as port
+from table(flatten(input => parse_json(system$whitelist()))) as t;
+
+```
+
+The regional account name will be in SNOWFLAKE_DEPLOYMENT field
```
Comment (Hamo69) [inline _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md:48]: > be in SNOWFLAKE_DEPLOYMENT
be in the SNOWFLAKE_DEPLOYMENT field

File: _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+Enter your app details and press continue
+
+<center><img src="/img/blog/snowflake-g-suite/04-Gsuite.png" /></center><br/>
+
+Copy the values of <code>SSO URL</code> , <code>Entity ID</code> and <code>Certificate</code> and click continue. You will need those later when setting up snowflake
+
+<center><img src="/img/blog/snowflake-g-suite/05-Gsuite.png" /></center><br/>
+
+Recently Snowflake has introduced a friendly name for the account, however in order to setup SSO with G-Suite you need snowflake auto generated account name and region. To retrieve your generated account name run below command in Snowflake.
```
Comment (Hamo69) [inline _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md:38]: casing issue again for "snowflake"

File: _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+<center><img src="/img/blog/snowflake-g-suite/09-Gsuite.png" /></center><br/>
+
+# Snowflake Setup:
+
+Now go to your Snowflake account and set saml_identity_provider on the account level. Replace the certificate, issuer and ssoUrl with values copied from G-suite app and run below commands.
+
+<div style="background-color: #fff3cd ; border-color: #ffeeba; color: #856404; border-radius: .25rem; padding: .75rem 1.25rem;"><strong>Note!</strong><br/>
+
+1. When entering the certificate into Snowflake please ensure the certificate is ALL ON ONE LINE (e.g. no carriage returns) along with remove the Begin and End Certificate tags<br/>
+2. issuer value will be from entity ID in your G-suite app<br/>
```
Comment (Hamo69) [inline _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md:82]: issuer should be Issuer

File: _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+3. run below commands using AccountAdmin role<br/>
```
Comment (Hamo69) [inline _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md:83]: run to Run

File: _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+Unfortunately automatic provisioning with G-suite does not work in Snowflake. Therefor you will have to manually create your users in snowflake. Run below command to create your users using their email addresses:
+
+<div style="background-color: #fff3cd ; border-color: #ffeeba; color: #856404; border-radius: .25rem; padding: .75rem 1.25rem;"><strong>Note!</strong><br/>You do not need to specify passwords for SSO users in snowflake</div> <br/>
+
+```sql
+CREATE USER "zainab.maleki@mechanicalrock.io";
+```
+
+Once all the above setup is completed, now you can test your integration using below URL:
```
Comment (Hamo69) [inline _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md:111]: > using below URL:
to using the below URL:

**Review by Hamo69 — APPROVED** [review]

File: _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+  alter account set sso_login_page = true;
+
+```
+
+<br/>
+
+# Verify your connection:
+
+Unfortunately automatic provisioning with G-suite does not work in Snowflake. Therefor you will have to manually create your users in Snowflake. Run below command to create your users using their email addresses:
```
Comment (aj-mechrock) [inline _posts/2021-07-05-snowflake-saml-auth-with-G-suite.md:103]: Therefore*

**Review by aj-mechrock — APPROVED** [review]
Lurve it! To the point and informative!!

**Pull request merged**